test/emqx_broker_SUITE.erl

```diff
@@ -206,6 +206,45 @@ t_stats_fun(_) ->
     ?assertEqual(2, emqx_stats:getstat('suboptions.count')),
     ?assertEqual(2, emqx_stats:getstat('suboptions.max')).
 
+t_worker_kill(_)->
+    TargetBroker = hd(supervisor:which_children(
+                  element(2,lists:keyfind(pool_sup, 1,
+                                          supervisor:which_children(emqx_broker_sup))))),
+    TargetBrokerPid = element(2, TargetBroker),
+    #{worker := Worker, worker_ref := Ref} = sys:get_state(TargetBrokerPid),
+    true = erlang:is_process_alive(Worker),
+    spawn(fun() -> link(Worker), exit(abnormal) end),
```

Comment:
@zmstone I decided not to try the 'link' approach due to this test.
because if the Fun that transaction is going to run with, happen to link some wild process which could cause worker killed and then broker process restart. 

The broker process is safe to link worker process only if broker trap exits, but I am unsure if it is ok to trap exit for the moment.

using a monitor is much safer than linking I think.

Reply:
ok.